File: src/IFWEMS.Client/src/app/core/models/compliance.models.ts
export interface TransactionCheckRequest {
  transactionReference: string;
  orgUnitId: string;
  supplierId?: string | null;
  contractId?: string | null;
  amount: number;
  invoiceReference?: string | null;
  purchaseOrderDateUtc?: string | null;
  commodityCode?: string | null;
}

export interface TransactionCheckResult {
  riskScore: number;
  riskRating: 'GREEN' | 'AMBER' | 'RED';
  failedRuleCodes: string[];
  recommendedAction: 'PASS' | 'REVIEW' | 'BLOCK';
}

export interface EmergencyOverrideRequest {
  transactionReference: string;
  orgUnitId: string;
  supplierId?: string | null;
  amount: number;
  emergencyCategory: string;
  reason: string;
  evidenceDocumentId: string;
  commodityCode?: string | null;
}

File: src/IFWEMS.Application/Compliance/ComplianceModels.cs
namespace IFWEMS.Application.Compliance;

public record TransactionCheckRequest(
    string TransactionReference,
    Guid OrgUnitId,
    Guid? SupplierId,
    Guid? ContractId,
    decimal Amount,
    string? InvoiceReference,
    DateTime? PurchaseOrderDateUtc,
    string? CommodityCode);

public record TransactionCheckResult(
    int RiskScore,
    string RiskRating,
    IReadOnlyList<string> FailedRuleCodes,
    string RecommendedAction);

public record EmergencyOverrideRequest(
    string TransactionReference,
    Guid OrgUnitId,
    Guid? SupplierId,
    decimal Amount,
    string EmergencyCategory,
    string Reason,
    Guid EvidenceDocumentId,
    string? CommodityCode);
